creusot/tests/should_succeed/syntax/06_logic_function_contracts.stdout

```diff
@@ -70,9 +70,35 @@ module C06LogicFunctionContracts_AllZero_Impl
     variant {Seq.length seq}
     
    = 
-    if Seq.length seq = 0 then
+    if pure {Seq.length seq = 0} then
       true
     else
-      Seq.get seq (Seq.length seq - 1) = 0 && all_zero (SeqExt.subsequence seq 0 (Seq.length seq - 1))
+      pure {Seq.get seq (Seq.length seq - 1) = 0} && all_zero (SeqExt.subsequence seq 0 (Seq.length seq - 1))
```

Comment:
I don't understand why we don't need to let bind the call to `SeqExt.subsequence` here, it's defined as just a `function`. I think this is another bizarre thing about how `let functions` work (we must be allowed to _call_ logic functions).